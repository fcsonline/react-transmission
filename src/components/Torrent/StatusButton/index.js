import React, { Component } from 'react';
import { themr } from 'react-css-themr';
import autobind from 'autobind-decorator';
import { observer } from 'mobx-react';

@themr('StatusButton')
@observer
class StatusButton extends Component {
  static defaultProps = {
    onToggle: () => {},
  }

  @autobind onClick(event) {
    this.props.onToggle(this.props.torrent.id);
  }

  render() {
    const { theme } = this.props;

    return (
      <button
        className={this.props.torrent.isStopped ? theme.statusResume : theme.statusPause}
        onClick={this.onClick}
      />
    );
  }
}

export default StatusButton;
