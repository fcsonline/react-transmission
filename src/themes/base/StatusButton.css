.statusButton {
  width: 14px;
  height: 14px;
  margin-left: 8px;
  background: url(./images/torrent-buttons.png);
  border: 0;
  padding: 0;
  cursor: pointer;
}

.statusButton:focus {
  border: 0;
  outline: none;
}

.statusPause {
  composes: statusButton;

  background-position: left top;
}

.statusPause:hover {
  background-position: left center;
}

.statusPause:active {
  background-position: left bottom;
}

.statusResume {
  composes: statusButton;

  background-position: center top;
}

.statusResume:hover {
  background-position: center center;
}

.statusResume:active {
  background-position: center bottom;
}
