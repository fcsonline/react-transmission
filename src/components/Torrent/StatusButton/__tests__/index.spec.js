import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import StatusButton from '../';

test('StatusButton resume', () => {
  const component = renderer.create(
    <StatusButton torrent={{isStopped: true}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StatusButton pause', () => {
  const component = renderer.create(
    <StatusButton torrent={{isStopped: false}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('StatusButton onToggle', () => {
  const toggleStatus = jest.fn();

  const component = mount(
    <StatusButton torrent={{id: 999, isStopped: false}} onToggle={toggleStatus} />
  );

  component.find('button').simulate('click');

  expect(toggleStatus).toHaveBeenCalledWith(999);
});
