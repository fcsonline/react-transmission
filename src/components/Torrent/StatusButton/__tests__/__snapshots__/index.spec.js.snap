// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusButton pause 1`] = `
<button
  className={undefined}
  onClick={[Function]}
/>
`;

exports[`StatusButton resume 1`] = `
<button
  className={undefined}
  onClick={[Function]}
/>
`;
